src/webgpu/shader/validation/parse/diagnostic.spec.ts

```diff
@@ -80,6 +80,11 @@ g.test('valid_params')
       .combine('rule', kSpecDiagnosticRules)
       .combine('type', kDiagnosticTypes)
   )
+  .beforeAllSubcases(t => {
+    if (t.params.rule === 'subgroup_uniformity') {
+      t.selectDeviceOrSkipTestCase('subgroups' as GPUFeatureName);
```

Comment:
It's valid to use unknown diagnostics.  I think this should not skip.
(I know testing unrecognized diagnostics is tested below for `warning_unknown_rule`)